PowerEditor/src/NppCommands.cpp

```diff
@@ -1246,11 +1252,10 @@ void Notepad_plus::command(int id)
 				dlgID = MARK_DLG;
 			_findReplaceDlg.doDialog(dlgID, _nativeLangSpeaker.isRTL());
 
-			const NppGUI & nppGui = (NppParameters::getInstance()).getNppGUI();
 			if (nppGui._fillFindFieldWithSelected)
 			{
-				_pEditView->getGenericSelectedText(str, strSize, nppGui._fillFindFieldSelectCaret);
-				if (lstrlen(str) <= FINDREPLACE_INSEL_TEXTSIZE_THRESHOLD)
+				if ((nppGui._inSelectionAutocheckThreshold == 0) ||
+					(lstrlen(str) < nppGui._inSelectionAutocheckThreshold))

```

Comment:
Selection Text to Find what should work independently. 
The condition of `_findReplaceDlg.setSearchText(str);` should be dissociated with **In selection**.
As a result, these lines should be restored.

Reply:
Sorry for not being clear.
What I mean is use the following code:
```cpp

				if (lstrlen(str) <= FINDREPLACE_INSELECTION_THRESHOLD_DEFAULT)
				{
					_findReplaceDlg.setSearchText(str);
				}
```
instead of 
```
				if ((nppGui._inSelectionAutocheckThreshold == 0) ||
					(lstrlen(str) < nppGui._inSelectionAutocheckThreshold))
				{
					_findReplaceDlg.setSearchText(str);
				}
```
to preserve the old behaviour of "selection to Find What" feature:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/9d05cb70-de27-4318-bca4-a6f71bae196b)

Reply:
@donho,

Could you please list the exact STR?
(Master vs. PR).

Thank you.

Reply:
@Yaron10 
Master : Ctrl-F copy the selected text (less 1024) into Find What field.
PR: when set the value to 8,  Ctrl-F does not copy the selected text (more than 8) into Find What field.

The ***Ctrl-F copy the selected text (less 1024) into Find What field*** should not be interfered by this PR. That's my point.

Reply:
@donho OK, will make this change in the next commit.

Reply:
@donho & @alankilborn,

Thank you for the clarification and work.

Selected "Close" not being set in "Find what" is indeed weird.
But the latest commit creates two major issues:

1. If auto-checking is disabled (0), 1024 is irrelevant and even 1025-selected should be set in "Find what".
\- Why should `if ((nppGui._inSelectionAutocheckThreshold == 0)` be removed?
2. If auto-checking is enabled, and the value is set to 5 - we have an inherent logic contradiction: you wouldn't want to search "Close" in the _selected "Close"_.
(And that logic/implementation is still valid for 1024).

Limiting the valid range to 600-1024 might be a solution.

Reply:
@Yaron10 

>But the latest commit creates two major issues

At this point, I just do what I'm told. :-)

Reply:
> Limiting the valid range to 600-1024 might be a solution.

Or:

> My opinion is not bother with that. If user want to set wired value like 6, 7, 8 or 9, it's their freedom to get wired behaviour (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14175#issuecomment-1739822133).

And the "freedom to get wired behaviour" should apply to _Selected "Close" not being set in "Find what"_ as well.

Reply:
@alankilborn @Yaron10 
The goal of this PR is to provide user the way to customize the length of auto-checking In selection and to disable the auto-checking In selection. 
Once user can do the above 2 things, they are on their own to set unreasonable value to have the inconsistent behaviour.
It's not critical for me such inconsistent behaviour because: 1. User will realize what is the reasonable value. 2. Even they don't, In selection can always be checked or unchecked. 3. In the worst case, they don't realize the reasonable value, and they don't check or uncheck conveniently, the search result is correct. The program is not buggy to me.
OTOH, change the behaviour of copying selected text into Find What due to this small enhancement is a big mistake to me - from the viewpoint of Notepad++ users.